python/tank_vendor/shotgun_base/utils.py

```diff
@@ -18,6 +18,34 @@
 from .log import get_shotgun_base_logger
 log = get_shotgun_base_logger()
 
+
+def with_cleared_umask(func):
+    """
+    Decorator which clears the umask for a method.
+
+    The umask controls default permissions for files
+    and folders. By setting the umask to zero, we can control
+    permissions.
+
+    Example use:
+
+    @with_cleared_umask
+    def my_io_method(param1, param2):
+
```

Comment:
Cool idea! I think for my requirements here, a decorator works and seems a little simpler. I always liked how it is very clear what is going on inside a decorated method. We can always add with support if needed later.